scintillometry/fourier/base.py

```diff
@@ -0,0 +1,345 @@
+    def ortho(self):
+        """Use orthogonal normalization.
+
+        If `True`, both forward and backward transforms are scaled by
+        1 / sqrt(n), where n is the size of time-domain array's transform
+        axis.  If `False`, forward transforms are unscaled and inverse ones
+        scaled by 1 / n.
+        """
+        return self._ortho
+
+    @property
+    def sample_rate(self):
+        """Rate of samples in the time domain."""
+        return self._sample_rate
+
+    @lazyproperty
+    def freq(self):
+        """FFT sample frequencies.
+
+        Uses `numpy.fft.fftfreq` for complex time-domain data, which returns,
+        for an array of length n and a time-domain ``sample_rate``,
+
+            f = [0, 1, ...,   n/2-1,     -n/2, ..., -1] * sample_rate / n
+
+        if n is even, and
+
+            f = [0, 1, ..., (n-1)/2, -(n-1)/2, ..., -1] * sample_rate / n
+
+        if n is odd.
+
+        For real time-domain data, `numpy.fft.rfftfreq` is used, which returns
+
+            f = [0, 1, ...,     n/2-1,     n/2] * sample_rate / n
+
+        if n is even, and
+
+            f = [0, 1, ..., (n-1)/2-1, (n-1)/2] * sample_rate / n
+
+        if n is odd.
+
+        If ``sample_rate`` is `None`, output is unitless.
+
+        Returns
+        -------
+        freqs : `~numpy.ndarray` or array of `~astropy.units.Quantity`
+            Sample frequencies.
+        """
+        sample_rate = self.sample_rate
+        if sample_rate is None:
+            sample_rate = 1.
+        a_length = self.data_format['time_shape'][self.axis]
+        if 'float' in self.data_format['time_dtype'].name:
+            return np.fft.rfftfreq(a_length, d=(1. / sample_rate))
+        return np.fft.fftfreq(a_length, d=(1. / sample_rate))
+
+    def __call__(self, a):
+        """Perform FFT.
+
+        To display the direction of the transform and shapes and dtypes of the
+        arrays, use `print` or `repr`.
+
+        Parameters
+        ----------
+        a : array_like
+            Input data.
+
+        Returns
+        -------
+        out : `~numpy.ndarray`
+            Transformed data.
+        """
+        return self._fft(a)
+
+    def inverse(self):
+        """Return inverse transform.
+
+        Returns
+        -------
+        inverse_transform : `~scintillometry.fourier.base.FFTBase` subclass
+            Returns a new instance of the calling class with reversed transform
+            direction.
+        """
+        return self.__class__(
+            direction=('forward' if self.direction == 'inverse'
+                       else 'inverse'))
+
+    def copy(self):
+        return self.__class__(direction=self.direction)
+
+    def __copy__(self):
+        return self.copy()
+
+    def __eq__(self, other):
+        # Assumes that class names are unique, which should be the case unless
+        # the user improperly initializes the class factory.
+        return (self.__class__.__name__ == other.__class__.__name__ and
```

Comment:
We do (eg. someone hoping to farm off expensive FFTs to GPUs or something), but not when using `==`.  Indeed, it might be advantageous for ``fft1 == fft2`` to return `True` even if the backend is different.  Removed the first check.